stream_color: Color picker popover improvements.

sayamsamal opened the pull request:
This PR is a follow-up for #32063, and makes the following changes:
- Updates custom color swatch icon to use rainbow gradient.
- Adds `buttons.css` file to `app.ts` bundle for production usage.
- Overrides the on-focus outline style in action-button and icon-button.
- Adds confirm button to the color picker popover.

Fixes:  [CZO Discussion](https://chat.zulip.org/#narrow/channel/101-design/topic/colorpicker.20migration.20and.20redesign/near/2013004)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

## Screenshots and screen captures:

### Screenshots:
| Header | Header |
|--------|--------|
| <img width="271" alt="Screenshot 2025-01-09 at 1 48 18 AM" src="https://github.com/user-attachments/assets/7ed13a68-f59d-4b68-aee9-bf594e3d01c9" /> | <img width="271" alt="Screenshot 2025-01-09 at 1 14 57 AM" src="https://github.com/user-attachments/assets/fbce3e81-607f-4f22-bcf4-d11d96196278" /> |
| <img width="271" alt="Screenshot 2025-01-09 at 1 48 02 AM" src="https://github.com/user-attachments/assets/211a8737-1178-49f6-9627-6cca5d99f929" /> | <img width="271" alt="Screenshot 2025-01-09 at 1 15 08 AM" src="https://github.com/user-attachments/assets/2f51116b-a5ff-4b44-aab6-72a19307b5ac" /> | 

### Screen captures:
![color-picker-popover-w-confirm](https://github.com/user-attachments/assets/5e386a08-a34f-4ba4-84f6-03043e5a81a5)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on web/src/bundles/app.ts:
```diff
@@ -57,6 +57,7 @@ import "../../styles/widgets.css";
 import "../../styles/print.css";
 import "../../styles/inbox.css";
 import "../../styles/color_picker.css";
+import "../../styles/buttons.css";
```
You probably want to import this earlier, as CSS precedence will generally prefer later things as an override at equal precedence, I think.

Probably just before `import "../../styles/components.css";` would be a reasonable placement.

timabbott commented on web/styles/buttons.css:
```diff
@@ -29,6 +29,11 @@
+    &:focus {
+        /* Override common button outline style set in zulip.css and dark_theme.css */
+        outline: none;
+    }
```
double-check this has the right precedence after my last suggestion, I guess.

sayamsamal commented:
## Updates
- Fixed the import of `buttons.css`, also had to modified the import precedence in `webpack.dev-assets.json` to match the same since webpack threw an error otherwise.
- Fixed enter key not working on the custom color picker popover option.
- Added better keyboard navigation for the color swatch grid, it now supports all the arrow keys and their viim-key counterparts.
- Fixed confirm button (check button) not being accessible via the keyboard.

https://github.com/user-attachments/assets/5879cd1e-9a03-4ca5-b1a2-d931f4cfc2a3

sayamsamal commented:
@timabbott I have addressed the feedback, and also added commits to improve the navigation in the color picker popover. Could you review this again?

alya commented:
Works for me in manual testing!

timabbott commented:
Looks great, merged, thanks @sayamsamal!

sayamsamal commented on web/templates/popovers/color_picker_popover.hbs:
```diff
@@ -13,24 +13,29 @@
-                    <label role="menuitemradio" class="color-swatch-label tippy-zulip-delayed-tooltip" for="color-{{hex_color}}" style="background-color: {{hex_color}};" aria-label="{{hex_color}}" data-tippy-content="{{hex_color}}" tabindex="0"></label>
+                {{#each stream_color_palette as |row|}}
+                    {{#each row as |hex_color|}}
+                        <input type="radio" id="color-{{hex_color}}" class="color-swatch-input" name="color-picker-select" data-swatch-color="{{hex_color}}" {{#if (eq hex_color ../stream_color )}}checked{{/if}} />
```
My bad, thanks for the fix!
